extensions/interactions/NumberWithUnits/directives/NumberWithUnits.js

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+oppia.directive('oppiaInteractiveNumberWithUnits', [
+  'HtmlEscaperService', 'UrlInterpolationService',
+  function(HtmlEscaperService, UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        onSubmit: '&',
+        // This should be called whenever the answer changes.
+        setAnswerValidity: '&'
+      },
+      templateUrl: UrlInterpolationService.getExtensionResourceUrl(
+        '/interactions/NumberWithUnits/directives/' +
+        'number_with_units_interaction_directive.html'),
+      controller: [
+        '$scope', '$attrs', 'FocusManagerService',
+        'NumberWithUnitsObjectFactory', 'NUMBER_WITH_UNITS_PARSING_ERRORS',
+        'WindowDimensionsService', 'EVENT_PROGRESS_NAV_SUBMITTED',
```

Comment:
Ditto.

Reply:
We need it to allow proper format in different sized windows. It is also used in almost all of the interactions as well.

Reply:
Done, removed.